manifests/install.pp

```diff
@@ -18,7 +18,7 @@
   $python = $::python::version ? {
     'system' => 'python',
     'pypy'   => 'pypy',
-    default  => "${python::version}",
+    default  => "${python::version}", # lint:ignore:only_variable_string
```

Comment:
I think you misunderstood what I am saying.  Placing the variable in quotes like that does NOT stringify it.  It is still an integer (assuming that the input is just digits) or a string (if the input is not just digits) either way.

Changing it to read
```
default => $python::version
```
is behaviorally identical to the quoted and braced version, just more stylistically correct.  Please read my previous line comments which demonstrate this.